Updating to version 3.0.0 of uswds-jekyll

line47 opened the pull request:
**Preview:** https://federalist-proxy.app.cloud.gov/preview/18f/federalist-uswds-template/sw-update-uswds/
- Added h3-h6 to the sample docs page
- Updated available config options for `config`, `header`, and `footer` YML files

#  V3.0.0 updates
## Added
- [Search.gov](https://search.gov/) as the option for search and no longer using [Jekyll pages api search](https://github.com/18F/jekyll_pages_api_search)
- Default 404 page
- Edit this page link option in the footer that can be shown or hidden via a config
- Support for Github Pages and [jekyll-remote-theme](https://github.com/benbalter/jekyll-remote-theme) plugin

## Updated
- USA banner to be it's own component that can be shown or hidden via a config
- USWDS to [version 1.4.5](https://github.com/uswds/uswds/releases/tag/v1.4.5)
- Updated USWDS naming from ["Standards" to "System"](https://designsystem.digital.gov/whats-new/updates/2018/01/17/changing-our-name/)
- Default footer code to align with the [USWDS 'medium' footer](https://components.designsystem.digital.gov/components/detail/footer--default.html)
- NPM task to exclude the USWDS SCSS files

## Removed 
- [Jekyll pages api search](https://github.com/18F/jekyll_pages_api_search) in favor of search.gov 
- Search results layout
- USWDS SCSS assets folder

Commit:
updating to version 3.0.0 of uswdw-jekyll

jseppi commented on _config.yml:
```diff
@@ -1,6 +1,27 @@
 title: Site title
 
+# GitHub information
```
I like all the explanatory comments in here and in the other config files

wslack replied:
agree

jseppi commented on _data/header.yml:
```diff
@@ -3,12 +3,21 @@
 # _config.yml).
-# title: Header title
+
+title: ""
```
this doesn't quite match with the comment above. Should it be commented out, or an empty string?

jseppi commented on _data/header.yml:
```diff
@@ -3,12 +3,21 @@
+title: ""
+
+# Used to show or hide the official usa banner at the top of the site.
+# Comment out to the following line to hide the usa banner.
```
or set to `false`? Which is preferred?

line47 replied:
Either works, I think we chose to comment out as that is the format for most all of the other config options

jseppi commented on _data/header.yml:
```diff
@@ -3,12 +3,21 @@
+usa_banner: true
+
+# this defines the type of header
+# header types can be 'basic' or 'extended'
```
maybe a succinct explanation of `basic` vs `extended` would be nice

line47 replied:
Do you think URLs to the component library would be helpful in the comments? 
https://components.designsystem.digital.gov/components/detail/header--basic.html
https://components.designsystem.digital.gov/components/detail/header--extended.html

jseppi replied:
oh, that's a good idea!

jseppi reviewed (changes requested):
This looks really good to me.

A couple pretty minor doc things to clean-up in `header.yml` if you wouldn't mind.

Commit:
updating comments to make more consistent

jseppi reviewed (approved):
🎉

line47 commented:
@wslack - like to get your 👀  on this before merging as well

wslack commented:
looking now

wslack commented on _config.yml:
```diff
@@ -1,6 +1,27 @@
 title: Site title
 
+# GitHub information
+# This is used for adding an edit this page link to the footer
```
are we going to put this in the footer later? If not, not sure we should add it.

line47 replied:
We can show the "edit this page" link by default, is that what you are suggesting?

wslack replied:
Yes! If we ask for the info in the config file, we should implement it or put the code in with a comment.

wslack reviewed (changes requested):
Love the documentation.

I think we need more work on the new footer. It seems a lot more unpredictable than the previous one at various widths.

wslack commented:
(to be clear - I'm referring to the footer on the docs page)

wslack commented on _data/footer.yml:
```diff
@@ -1,26 +1,65 @@
 logos:
-  - src: /assets/img/us_seal.png
-    alt: Your agency name
+  - src: https://18f.gsa.gov/assets/img/logos/18f-logo.svg
```
why not use an internal 18F logo and an external GSA logo to show both ways?

line47 replied:
We can add both logos by default - that's an easy update.

wslack replied:
Let's do that - to show both ways to link an image.

line47 commented:
@wslack can you explain a bit more about how the footer is unpredictable?
We moved from a custom footer to using the USWDS medium (default) footer, maybe there are a few things that could be addressed upstream?

https://components.designsystem.digital.gov/components/detail/footer--default.html

wslack commented:
@line47 the behavior of the footer at various page widths is wonky. It jumps up and down and gets out of sorts.

wslack commented:
I can't figure out what I was referring to. I don't like how the footer looks on the documentation page (it appears too similar to the menu) but I don't understand "unprediable.' Maybe was related to using a large screen at the office. I think you can discard.

Commit:
adding git link in footer

line47 commented:
Updated to add a the GitHub link to the footer, will need to add functionality for multiple logos in the footer to the uswds-jekyll. At the moment the images are right against each other and need some visual separation. I've added an issue for this https://github.com/18F/uswds-jekyll/issues/127 

@wslack I think this is ready for another pass 👍

wslack commented:
That image for the GSA logo doesn't work for me. @line47

wslack commented on _data/footer.yml:
```diff
@@ -1,26 +1,67 @@
 logos:
-  - src: /assets/img/us_seal.png
-    alt: Your agency name
+  - src: /assets/img/gsa-logo-dark.jpg
```
Looks like the file needs to be dropped into https://github.com/18F/federalist-uswds-template/tree/master/assets/img

line47 replied:
Thanks for that - I've added the second logo (GSA). We should be good to go!

Commit:
adding gsa logo to footer

Commit:
Updating links so there a no 404s

Commit:
Merge branch 'master' into sw-update-uswds
